src/firebase/saga.ts

```diff
@@ -67,6 +77,7 @@ function* initializeFirebase() {
 
         yield call(initializeAuth, firebase, address)
         yield put(firebaseAuthorized())
+        yield waitForVisitHome()
         yield call(initializeCloudMessaging, firebase, address)
```

Comment:
Are there any side effects/other consequences of waiting to call `initializeCloudMessaging` until the home screen is seen? Just scanning, that function seems to do a lot. Is the only real consequence that push notifications won't be enabled until the home screen is reached?

Reply:
I had the same thought, and my initial reaction was well, this still happens every time a user visits home, which is most of the time when they open the app.

But I'm a bit uneasy about this, particularly as we move things around in the app-- if there wasn't a singular homepage anymore, for example, and some users spent lots of time away from home, this could cause problems in a very unexpected and hard to detect way. There's some advice in the docstring for `app.messaging().onTokenRefresh` (see [line 207 in firebase.ts](https://github.com/valora-inc/wallet/blob/0b4e518cc46dc98bbc19aca9bbd81222ca904981/src/firebase/firebase.ts#L189)) that says to run it on app open. 

I'm looking into a way to be more precise on when to wait for the homescreen, which might be better to do exactly [here](https://github.com/valora-inc/wallet/blob/0b4e518cc46dc98bbc19aca9bbd81222ca904981/src/firebase/firebase.ts#L154), since it only happens if the user hasn't already enabled messaging. If they have enabled messaging previously, all the other stuff in that initializeCloudMessaging function can still take place.

Reply:
k just updated the PR